lib/galaxy/schema/workflows.py

```diff
@@ -223,3 +385,642 @@ class StoredWorkflowDetailed(StoredWorkflowSummary):
+        description="Upgrade messages for each step in the workflow.",
+    )
+    # TODO - can this be modeled further? see manager method _workflow_to_dict_editor
+    report: Dict[str, Any] = Field(
+        ...,
+        title="Report",
+        description="The reports configuration for the workflow.",
+    )
+    comments: List[WorkflowCommentModel] = Field(
+        ...,
+        title="Comments",
+        description="Comments on the workflow.",
+    )
+    annotation: WorkflowAnnotationField
+    license: Optional[str] = Field(
+        ...,
+        title="License",
+        description="SPDX Identifier of the license associated with this workflow.",
+    )
+    creator: WorkflowCreator
+    source_metadata: Optional[Dict[str, Any]] = Field(
+        ...,
+        title="Source Metadata",
+        description="Metadata about the source of the workflow",
+    )
+    steps: Dict[int, WorkflowDictEditorStep] = Field(
+        ...,
+        title="Steps",
+        description="Information about all the steps of the workflow.",
+    )
+
+
+class WorkflowDictRunSummary(WorkflowDictExtendedBaseModel):
+    id: str = Field(
+        ...,
+        title="ID",
+        description="The encoded ID of the stored workflow.",
+    )
+    history_id: Optional[str] = Field(
+        None,
+        title="History ID",
+        description="The encoded ID of the history associated with the workflow.",
+    )
+    step_version_changes: List[Union[str, Dict[str, Any]]] = Field(
+        ...,
+        title="Step Version Changes",
+        description="Version changes for the workflow steps.",
+    )
+    has_upgrade_messages: bool = Field(
+        ...,
+        title="Has Upgrade Messages",
+        description="Whether the workflow has upgrade messages.",
+    )
+    workflow_resource_parameters: Optional[Dict[str, Any]] = Field(
+        ...,
+        title="Workflow Resource Parameters",
+        description="The resource parameters of the workflow.",
+    )
+    steps: List[Union[WorkflowDictRunToolStep, WorkflowDictRunStep]] = Field(
+        ...,
+        title="Steps",
+        description="Information about all the steps of the workflow.",
+    )
+
+
+class WorkflowDictExportSummary(WorkflowDictBaseModel):
+    a_galaxy_workflow: Literal["true"] = Field(
+        # a_galaxy_workflow: str = Field(
+        ...,
+        title="A Galaxy Workflow",
+        description="Whether this workflow is a Galaxy Workflow.",
+    )
+    version: Optional[int] = Field(
+        None,
+        title="Version",
+        description="The version of the workflow represented by an incremental number.",
+    )
+    format_version: Literal["0.1"] = Field(
+        # format_version: str = Field(
+        ...,
+        alias="format-version",
+        title="Format Version",
+        description="The version of the workflow format being used.",
+    )
+    annotation: WorkflowAnnotationField
+    tags: Union[TagCollection, Literal[""]] = Field(
```

Comment:
```suggestion
    tags: TagCollection = Field(
```
this should always be a list, if not we should fix that.